test/init.js

```diff
@@ -55,6 +56,13 @@ global.getDataSource = global.getSchema = function(useUrl) {
   return db;
 };
 
+global.resetDataSourceClass = function(ctor) {
+  DataSource = ctor || juggler.DataSource;
+  var promise = db ? db.disconnect() : Promise.resolve();
```

Comment:
Have you thought about returning `Promise.reject` with a descriptive message if `db` is not truthy?

Reply:
It's a valid situation that `db` is not defined. This typically happens when `resetDataSourceClass` is called for the first time, for example when you run `mocha deps/juggler-v3/test.js`. 

Here is the logic of this function:
 - If there is an existing `db` instance left by previous tests, then dispose it (call `disconnect`). We don't want to have a growing number of datasource instances, because each datasource is usually maintaining one or more open connections to the database.
 - (Re)set the datasource class to the new value.